config/schema/schema.json

```diff
@@ -0,0 +1,298 @@
+{
+  "type": "object",
+  "properties": {
+    "indexing_sources": {
+      "type": "object",
+      "properties": {
+        "base_data_bucket": {
+          "type": "string"
+        },
+        "base_data_file": {
+          "type": "string"
+        },
+        "subgenre_lut_file": {
+          "type": "string"
+        },
+        "thematic_lut_file": {
+          "type": "string"
+        },
+        "models_to_index": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "base_data_bucket",
+        "base_data_file",
+        "subgenre_lut_file",
+        "thematic_lut_file",
+        "models_to_index"
+      ]
+    },
+    "databackend": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "opensearch": {
+      "type": "object",
+      "properties": {
+        "host": {
+          "type": "string"
+        },
+        "port": {
+          "type": "string"
```

Comment:
Ich nehme an, hier kann nicht "number" stehen, da wir die Notation mit den Environment-Variablen haben. Ist an der Stelle halt etwas schade. Könnte man es denn irgendwie hinkriegen, dass die Variable interpoliert wird, bevor die Validation abläuft?

Reply:
korrekt!
Out of the box ist das nicht möglich.
Wenn müsste man ne custom Lösung bauen, die zuerst die config "rendert" und dann erst die Validierung macht
Und rein infrastrukturell stell ich mir das schwierig vor, weil ne CI/CD ja z.B. keinen Zugriff auf die Variablen hat, die sind ja dann erst zur Laufzeit bekannt.